packages/flutter/lib/src/widgets/autofill.dart

```diff
@@ -225,3 +228,142 @@ class _AutofillScope extends InheritedWidget {
+  const AutofillContextLifecycleAction({
+    Key key,
+    Widget child,
+    this.lifecycleDelegate,
+  }): super(key: key, child: child);
+
+  /// The [AutofillContextLifecycleDelegate] that descendant
+  final AutofillContextLifecycleDelegate lifecycleDelegate;
+
+  /// The [AutofillContextLifecycleDelegate] to fall back to, when there's no
+  /// enclosing [AutofillContextLifecycleAction] in the context.
+  ///
+  /// This value can be changed to modify the global configuration. Changes
+  /// apply immediately: the new [globalLifecycleDelegate] takes effect starting
+  /// from the next monitored lifecycle event.
+  ///
+  /// Must not be null.
+  static AutofillContextLifecycleDelegate get globalLifecycleDelegate => _globalLifecycleDelegate;
+  static AutofillContextLifecycleDelegate _globalLifecycleDelegate = const AutofillContextLifecycleDelegate();
+  static set globalLifecycleDelegate(AutofillContextLifecycleDelegate newDelegate) {
+    assert(newDelegate != null);
+    if (_globalLifecycleDelegate != newDelegate)
+      _globalLifecycleDelegate = newDelegate;
+  }
+
+  /// The [AutofillContextLifecycleDelegate] from the closest
+  /// [AutofillContextLifecycleAction] widget that encloses the given [context],
+  /// or the default configuration [AutofillContextLifecycleAction.globalLifecycleDelegate]
+  /// if the given [context] does not contain an [AutofillContextLifecycleDelegate].
+  static AutofillContextLifecycleDelegate of(BuildContext context) {
+    return context?.dependOnInheritedWidgetOfExactType<AutofillContextLifecycleAction>()?.lifecycleDelegate ?? globalLifecycleDelegate;
+  }
+
+  // Changing delegates shouldn't make anything other than the widget itself
+  // rebuild.
+  @override
+  bool updateShouldNotify(covariant AutofillContextLifecycleAction oldWidget) => false;
+}
+
+/// Configures whether the current autofill context should be saved or discarded
+/// , or no action should be taken at all, when [Navigator]s and [Form]s
+/// configured by this [AutofillContextLifecycleDelegate] observes certain types
+/// of events.
+///
+/// Typically retrieved using [AutofillContextLifecycleAction.of].
+///
+/// The default implementation signals the platform to save the current autofill
+/// context (before discarding it), when any of the affected [Form]s is saved,
+/// or any of the affected [Navigator]s pushes/pops opaque [ModalRoute], or the
+/// current route is replaced with an opaque [ModalRoute].
+class AutofillContextLifecycleDelegate {
+  /// Creates a delegate object that can be used to clean up the current
+  /// autofill context at proper times, for example, when [Route]s change or
+  /// when [Form]s are saved.
+  const AutofillContextLifecycleDelegate();
+
+  /// The action needs to be taken after [FormState.save] is called on
+  /// [savedForm].
+  ///
+  /// The default implementation saves the current autofill context.
+  void onFormSave(FormState savedForm) {
```

Comment:
question: why does the function receive savedForm? I remember we discussed in the future we might want to send more parameters with save. Is it related to discussion?

Reply:
The `FormState` here is just a bit of additional information that helps the developer determine what needs to be done when this particular form gets saved. The framework doesn't really send the information to the engine.